src/Data/CAS/RocksDB.hs

```diff
@@ -301,6 +305,44 @@ tableDelete db k = R.delete
     (encKey db k)
 {-# INLINE tableDelete #-}
 
+-- -------------------------------------------------------------------------- --
+-- Batches
+
+-- | A single update operation for a 'RocksDbTable'
+--
+data RocksDbUpdate
+    = forall k v . RocksDbDelete
+        { _rocksDbUpdateTable :: !(RocksDbTable k v)
+        , _rocksDbUpdateKey :: !k
+        }
+    | forall k v . RocksDbInsert
+        { _rocksDbUpdateTable :: !(RocksDbTable k v)
+        , _rocksDbUpdateKey :: !k
+        , _rocksDbUpdateValue :: !v
+        }
+
+rocksDbUpdateDb :: RocksDbUpdate -> R.DB
+rocksDbUpdateDb (RocksDbDelete t _) = _rocksDbTableDb t
+rocksDbUpdateDb (RocksDbInsert t _ _) = _rocksDbTableDb t
+{-# INLINE rocksDbUpdateDb #-}
+
+-- | Atomically execute a batch of rocks db updates.
+--
+-- | ASSUMPTION: All tables in the batch operations belong to the same rocks db
+-- instance.
+--
+updateBatch :: [RocksDbUpdate] -> IO ()
```

Comment:
Use a `Traversable` rather than list

Reply:
Switching to traversable would provide relatively little benefit but make the code more complex. I think, I'll leave it to the caller to apply `toList` if needed.

I pushed a new version that checks the assumption that all all operations in the batch refer to the same DB.